queues/nats/nats_test.go

```diff
@@ -12,8 +14,11 @@ import (
 )
 
 func TestTransport(t *testing.T) {
-	transport := New()
-	vicetest.Transport(t, transport)
+	new := func() vice.Transport {
+		return New()
+	}
+
+	vicetest.Transport(t, new)
```

Comment:
only if we change all constructor functions (New) to return the interface type instead of the concrete type:

```
# github.com/matryer/vice/queues/nats
./nats_test.go:19:20: cannot use New (type func() *Transport) as type func() vice.Transport in argument to vicetest.Transport
```

or am i wrong here ?